src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -555,6 +555,37 @@ public void Dispose()
                 }
             }
         }
+
+        private Assembly? ResolveSatelliteAssembly(AssemblyName assemblyName)
+        {
+            string? cultureName = assemblyName.CultureName;
+
+            if (cultureName == null || cultureName.Length == 0)
+                return null;
+
+            if (assemblyName.Name == null)
+                return null;
+
+            AssemblyName parentAssemblyName = new AssemblyName(assemblyName.Name);
+
+            Assembly? parentAssembly = LoadFromAssemblyName(parentAssemblyName);
+
+            if (parentAssembly == null)
+                return null;
+
+            // ResolveSatelliteAssembly should always be called on the ALC which loaded parentAssembly
+            Debug.Assert(this == GetLoadContext(parentAssembly));
```

Comment:
If this is true it should be in theory possible to pass the parent assembly from the caller... I assume that's just too complicated.

Will the above `LoadFromAssemblyName` invoke ALC code, or will we get it from a runtime cache early on?

Reply:
> pass the parent assembly from the caller … too complicated

It could be done. This seemed simpler. 

The managed caller is passing the parent assembly. I could rework the native interfaces to keep this information. 

> will we get it from a runtime cache

It should be returned from the runtime cache
We would get more benefit from optimizing the cache lookup path.

Reply:
If it's gonna come from the cache, then this is perfectly fine... please ignore :-)

Reply:
> If this is true

This would be true for normal usage, but not always true.  In case an app tries to manually load a random satellite assembly w/o loading its parent, this assumption would be invalid.  I added back the original logic to get the ALC for the parent assembly to handle this rare case.